### PR TITLE
Fix OSX builds

### PR DESCRIPTION
**Pull request by Tobias-Fischer**
See https://github.com/ros/geometry2/pull/502 for reasoning. This was confirmed to work in the RoboStack project: https://github.com/RoboStack/ros-noetic/issues/58

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -7,11 +7,19 @@ set(Python_ADDITIONAL_VERSIONS "${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}"
...
+    LIBRARIES rospack ${PYTHON_LIBRARIES}
+    DEPENDS Boost TinyXML2
+  )
+else()
+  catkin_package(
+    INCLUDE_DIRS include
+    LIBRARIES rospack
+    DEPENDS Boost TinyXML2
+ )
+endif()
```
Comment (jacobperron) [inline CMakeLists.txt]: To help reduce duplication, I think it would be nicer to do something like this:

```cmake
set(package_libraries rospack)
if(NOT APPLE)
  list(APPEND package_libraries ${PYTHON_LIBRARIES}
endif()

catkin_package(
  INCLUDE_DIRS include
  LIBRARIES ${package_libraries}
  DEPENDS Boost TinyXML2
)
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -29,7 +37,17 @@ add_library(rospack
   src/rospack_cmdline.cpp
   src/utils.cpp
 )
-target_link_libraries(rospack ${TinyXML2_LIBRARIES} ${Boost_LIBRARIES} ${PYTHON_LIBRARIES})
+
+if(NOT APPLE)
+  target_link_libraries(rospack ${TinyXML2_LIBRARIES} ${Boost_LIBRARIES} ${PYTHON_LIBRARIES})
+else()
+  target_link_libraries(rospack ${TinyXML2_LIBRARIES} ${Boost_LIBRARIES})
+endif()
```
Comment (jacobperron) [inline CMakeLists.txt]: Same here, something like

```cmake
set(link_libraries ${TinyXML2_LIBRARIES} ${Boost_LIBRARIES})
if(NOT APPLE)
  list(APPEND link_libraries ${PYTHON_LIBRARIES})
endif()
target_link_libraries(rospack ${link_libraries})
```

**Review by jacobperron — COMMENTED** [review]
Some minor suggestions, otherwise LGTM!

**Comment by Tobias-Fischer** [comment]
Thanks for looking at that @jacobperron. I incorporated your suggestion. The link to PYTHON_LIBRARIES was actually needed, so I removed this part from the PR. We confirmed that rospack now works as intended in the RoboStack project: https://github.com/RoboStack/ros-noetic

File: CMakeLists.txt [hunk 3]
```diff
@@ -29,6 +34,7 @@ add_library(rospack
   src/rospack_cmdline.cpp
   src/utils.cpp
 )
+
```
Comment (jacobperron) [inline CMakeLists.txt]: nit: remove unrelated whitespace change to keep diff minimal

File: CMakeLists.txt [hunk 4]
```diff
@@ -7,9 +7,14 @@ set(Python_ADDITIONAL_VERSIONS "${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}"
 find_package(PythonLibs "${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}" REQUIRED)
 find_package(TinyXML2 REQUIRED)
 
+set(package_libraries rospack)
+if(NOT APPLE)
+  list(APPEND package_libraries ${PYTHON_LIBRARIES}
```
Comment (jacobperron) [inline CMakeLists.txt]:
```suggestion
  list(APPEND package_libraries ${PYTHON_LIBRARIES})
```

**Comment by Tobias-Fischer** [comment]
Done.

**Comment by Tobias-Fischer** [comment]
Thanks for merging @jacobperron!
